lib/shared/markdown.js

```diff
@@ -5,7 +5,7 @@ const paragraphRegex = /^((?:[^\n]*)(?:\n|$))/;
 const blockQuoteRegex = /^( *>[^\n]+(?:\n[^\n]+)*)(?:\n\n|$)/;
 const headingRegex = /^ *(#{1,6}) ([^\n]+?)#* *(?:\n|$)/;
 const codeBlockRegex = /^(?: {4}[^\n]+\n*)+(?:\n *)/;
-const fenceRegex = /^(`{3,}|~{3,})(?:(.*$)\n)?([\s\S]*)\1(?:\n|$)/m;
+const fenceRegex = /^(`{3,}|~{3,})(?:(.*$)\n)?([\s\S]*)\1(?:\n|$)/;
```

Comment:
I've changed what you suggested a little bit, and added non capturing group at the end, to not include empty line that's left after \``` . I also wanted to use `m` to achieve the same with empty line at the begining, but then I've realized it's breaking in some cases